python/cuml/cuml/neighbors/weights.py

```diff
@@ -0,0 +1,69 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: Apache-2.0
+
+import cupy as cp
+
+
+def compute_weights(distances, weights):
+    """
+    Compute weights from distances for KNN algorithms.
+
+    Parameters
+    ----------
+    distances : cupy.ndarray or array-like
+        The input distances (n_samples, k).
+
+    weights : {'uniform', 'distance'} or callable
+        The kind of weighting used.
+
+    Returns
+    -------
+    weights_arr : cupy.ndarray
+        Weights of shape (n_samples, k).
+        For uniform weights, returns array of 1/k (normalized).
+        For distance weights, returns raw inverse distances (not normalized).
+        For callable, returns raw result of callable(distances) (not normalized).
+    """
+    # Convert to cupy array if needed and ensure 2D
+    if not isinstance(distances, cp.ndarray):
+        distances = cp.asarray(distances)
```

Comment:
`cp.asarray` is a no-op if it's already a cupy array, no need for the branch

```suggestion
    distances = cp.asarray(distances)
```

Reply:
Addressed in [f6afbad](https://github.com/rapidsai/cuml/pull/7289/commits/f6afbad05edfc8d67cca9da57f1e6ed42f00e93a)